File: Weather-app/Models/HourlyWeatherData.m
//
//  HourlyWeatherData.m
//  Weather-app
//
//  Created by Reve on 11/14/22.
//

#import "HourlyWeatherData.h"

@implementation HourlyWeatherData

-(nullable instancetype) initWithDictionary:(nullable NSDictionary*) dict {
    if(self = [super init]) {
        self.time = dict[@"time"];
        self.temperature_2m = dict[@"temperature_2m"];
    }
    return self;
}
@end

File: WeatherApp/Models/HourlyWeatherData.swift
//
//  HourlyWeatherData.swift
//  Weather-app
//
//  Created by Reve on 11/20/22.
//

import Foundation

class HourlyWeatherData: NSObject, Codable {
    
    let time: [String]
    let temperature_2m: [String]
    
    enum CodingKeys: String, CodingKey {
        case time = "time"
        case temperature_2m = "temperature_2m"
    }
    
    required public init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        time = try container.decode([String].self, forKey: .time)
        temperature_2m = try container.decode([String].self, forKey: .temperature_2m)
        super.init()
    }
}
